test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NuGetUITestCase.cs

```diff
@@ -928,5 +941,62 @@ public async Task VerifyRestorePackageByBuilding()
             // Assert
             Directory.Exists(installedPackageFolderPath).Should().BeTrue();
         }
+
+
+        [DataTestMethod]
+        [Timeout(DefaultTimeout)]
+        public void VerifyPackageNotRestoredAfterDisablingPackageRestore()
+        {
+            // Arrange
+            EnsureVisualStudioHost();
+
+            SolutionService solutionService = VisualStudio.Get<SolutionService>();
+            solutionService.CreateEmptySolution("TestSolution", _pathContext.SolutionRoot);
+            solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.IOSLibraryApp, "TestProject");
+            VisualStudio.ClearOutputWindow();
+            solutionService.SaveAll();
+
+            // Act
+            _pathContext.Settings.DisableAutoRestore();
+            CommonUtility.RestoreNuGetPackages(VisualStudio, Logger);
+            VisualStudio.ObjectModel.Shell.ToolWindows.ErrorHub.ShowErrors();
+
+            // Assert
+            VisualStudio.AssertErrorListContainsSpecificError("NuGet restore is currently disabled.");
+        }
+
+        [DataTestMethod]
+        [Timeout(DefaultTimeout)]
+        public async Task VerifyPackageNotRestoredAfterDisablingAutomaticInPackageRestoreSection()
+        {
+            // Arrange
+            NuGetApexTestService nugetTestService = GetNuGetTestService();
+
+            await CommonUtility.CreatePackageInSourceAsync(_pathContext.PackageSource, TestPackageName, TestPackageVersionV1);
+
+            SolutionService solutionService = VisualStudio.Get<SolutionService>();
+            solutionService.CreateEmptySolution("TestSolution", _pathContext.SolutionRoot);
+            ProjectTestExtension project = solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.ConsoleApplication, "TestProject");
+            VisualStudio.ClearOutputWindow();
+            solutionService.SaveAll();
+
+            CommonUtility.OpenNuGetPackageManagerWithDte(VisualStudio, Logger);
+            NuGetUIProjectTestExtension uiwindow = nugetTestService.GetUIWindowfromProject(project);
+            uiwindow.InstallPackageFromUI(TestPackageName, TestPackageVersionV1);
+
+            // Act
+            _pathContext.Settings.DisableAutomaticInPackageRestoreSection();
+
+            var installedPackageFolderPath = Path.Combine(_pathContext.PackagesV2, "Contoso.A.1.0.0");
+            Directory.Exists(installedPackageFolderPath).Should().BeTrue();
+            Directory.Delete(installedPackageFolderPath, true);
+            CommonUtility.WaitForDirectoryNotExists(installedPackageFolderPath);
```

Comment:
Why do we need this here?

Reply:
The test case is to verify that when building a project with automatic disabled, the deleted packages don't back, so we added this code.

Reply:
I see. But then you run build and assert the same thing. 

My concern is that this just adds a 10s overhead. 
I'd just remove it. There's no action that happens. 
We're expecting it to not exist even after build, so it'd fail anyways, and we'd figure out where the problem is.

Reply:
Thanks for your suggestion, we have removed it. Could you please help to review and approve it, thanks.